cli/cmd/upgrade.go

```diff
@@ -263,6 +267,11 @@ func (options *upgradeOptions) validateAndBuild(stage string, k kubernetes.Inter
 	}
 
 	if options.identityOptions.trustPEMFile != "" {
+
+		if configs.Global.IdentityContext.Scheme == string(corev1.SecretTypeTLS) {
+			return nil, nil, errors.New("cannot update trust roots if you are using external cert management solution")
```

Comment:
Yes this all makes sense. The reason I got a bit hesitant about this part is that if we are using external management solution the `linkerd-identity-issuer` secret has a different format, one where the roots are contained there as well. So I was thinking, in that case we need to update the roots in both places and that kind of made me thing that later on we might want to introduce a functionality where modifying this secret updated the config map instead of providing that through the upgrade. But you are right, for the time being we need to go along and allow all this.

Reply:
Or do we assume that they modify it themselves in the linkerd-identity-issuer secret and use the upgrade command to modify it in the config. The reason I am hesitant about that is the fact that as far as I take it the mantra of the upgrade command is that it should not be side-effecting - namely it should not modify any resources without reflecting that in the output that it produces. And it cannot reflect this in the output because when we are using external issuer we do not produce any issuer secret yaml.

Reply:
Added a commit that does exactly what described above + tests. Feel free to take a quick look before I merge that

Reply:
My assumption is that this flag will only update the trust root in the linkerd configmap (**not** update the ca in the external issuer secret).  This reflects the division of responsibilities:
* the external issuer is responsible for updating the issuer secret
* Linkerd is responsible for the linkerd config map

Any changes to the ca in the external issuer secret need to be manually propagated into the Linkerd config map by running upgrade with this flag.